jetbrains-core/src/software/aws/toolkits/jetbrains/services/telemetry/DefaultTelemetryPublisher.kt

```diff
@@ -35,8 +34,8 @@ class DefaultTelemetryPublisher(
     private val os: String = SystemInfo.OS_NAME,
     private val osVersion: String = SystemInfo.OS_VERSION
 ) : TelemetryPublisher {
-    override fun publish(metricEvents: Collection<MetricEvent>): Boolean = ApplicationManager.getApplication().executeOnPooledThread<Boolean> {
-        try {
+    override suspend fun publish(metricEvents: Collection<MetricEvent>) {
+        withContext(Dispatchers.IO) {
```

Comment:
Style question: should we do `= withContext` since the whole function is withContext?

Reply:
yes and when the function has an explicit return type, we can - but you can't when the function is effectively a `Unit`.